test/shared.js

```diff
@@ -144,4 +144,54 @@ X-Poedit-SourceCharset: UTF-8`;
       expect(folded.length).to.equal(3);
     });
   });
+
+  describe('parseNPluralFromHeadersSafely', () => {
+    it('should return parsed value', () => {
+      const headers = { 'Plural-Forms': 'nplurals=10; plural=n' };
+      const nplurals = parseNPluralFromHeadersSafely(headers);
+
+      expect(nplurals).to.equal(10);
+    });
+
+    it('should return parsed value (missing plural declaration)', () => {
+      const headers = { 'Plural-Forms': 'nplurals=10' };
+      const nplurals = parseNPluralFromHeadersSafely(headers);
+
+      expect(nplurals).to.equal(10);
+    });
+
+    it('should return fallback value ("Plural-Forms" header is absent)', () => {
+      const nplurals = parseNPluralFromHeadersSafely();
+
+      expect(nplurals).to.equal(1);
+    });
+
+    it('should return fallback value (nplurals is not declared)', () => {
+      const headers = { 'Plural-Forms': '; plural=n' };
+      const nplurals = parseNPluralFromHeadersSafely(headers);
+
+      expect(nplurals).to.equal(1);
+    });
+
+    it('should return fallback value (nplurals is set to zero)', () => {
+      const headers = { 'Plural-Forms': 'nplurals=0' };
+      const nplurals = parseNPluralFromHeadersSafely(headers);
+
+      expect(nplurals).to.equal(1);
+    });
+
+    it('should return fallback value (nplurals is set to negative value)', () => {
```

Comment:
Should it return fallback value in this case or rather throw an error? I would opt for the latter

Reply:
Yeah, I also have doubts whether `nplurals` parsing function should try parsing it safely and fallback to default value or throw an error. I decided to go with safe variant and leave it for Sam to judge. @smhg what do you think?

Reply:
@KubaZ @vkhytskyi-allegro Is a negative value any different from something like 'foo' (see next test)? Or do did you consider them to be different because of the `parseInt()` return value check (inside `parseNPluralFromHeadersSafely`)?

Reply:
I'd stick with the fallback approach. Validating headers might be a good thing, but I think we shouldn't start only validating (= throwing an error) one part (`nplurals`) of one header (`Plural-Forms`).